testing/e2e/suites/osm.go

```diff
@@ -2,18 +2,37 @@ package suites
 
 import (
 	"context"
+	"fmt"
 
+	appManifests "github.com/Azure/aks-app-routing-operator/pkg/manifests"
 	"github.com/Azure/aks-app-routing-operator/testing/e2e/infra"
 	"github.com/Azure/aks-app-routing-operator/testing/e2e/manifests"
 	corev1 "k8s.io/api/core/v1"
 	netv1 "k8s.io/api/networking/v1"
 	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 var (
 	osmNs = make(map[string]*corev1.Namespace)
 )
 
+func deleteNginxPods(config *rest.Config) error {
+	c, err := client.New(config, client.Options{})
+	if err != nil {
+		return fmt.Errorf("creating client: %w", err)
+	}
+
+	podLabels := appManifests.GetTopLevelLabels()
+	podLabels["app.kubernetes.io/component"] = "ingress-controller"
+	err = c.DeleteAllOf(context.Background(), &corev1.Pod{}, client.InNamespace(manifests.ManagedResourceNs), client.MatchingLabels(podLabels))
```

Comment:
you should pretty much always be using assignment in the if statement. that is the go pattern

Reply:
done